src/backend/marsha/core/api/video.py

```diff
@@ -820,3 +822,102 @@ def initiate_upload(self, request, pk=None):
         Thumbnail.objects.filter(pk=pk).update(upload_state=defaults.PENDING)
 
         return Response(presigned_post)
+
+
+class SharedLiveMediaViewSet(ObjectPkMixin, viewsets.ModelViewSet):
+    """Viewset for the API of the SharedLiveMedia object."""
+
+    permission_classes = [permissions.NotAllowed]
+    queryset = SharedLiveMedia.objects.all()
+    serializer_class = serializers.SharedLiveMediaSerializer
+
+    def get_permissions(self):
+        """Instantiate and return the list of permissions that this view requires."""
+        if self.action in ["create", "list"]:
+            permission_classes = [
+                permissions.IsTokenInstructor
+                | permissions.IsTokenAdmin
+                | permissions.IsParamsVideoAdminThroughOrganization
+                | permissions.IsParamsVideoAdminThroughPlaylist
+            ]
+        elif self.action in ["retrieve"]:
+            permission_classes = [
+                permissions.IsTokenResourceRouteObjectRelatedVideo
+                | permissions.IsRelatedVideoPlaylistAdmin
+                | permissions.IsRelatedVideoOrganizationAdmin
+            ]
+        else:
+            permission_classes = [
+                permissions.IsTokenResourceRouteObjectRelatedVideo
+                & (permissions.IsTokenInstructor | permissions.IsTokenAdmin)
+                | permissions.IsRelatedVideoPlaylistAdmin
+                | permissions.IsRelatedVideoOrganizationAdmin
+            ]
+        return [permission() for permission in permission_classes]
+
+    def list(self, request, *args, **kwargs):
+        """List shared live media through the API."""
+        queryset = self.get_queryset().none()
```

Comment:
Because this is the way you get the query_set in an action ? You define a query_set for your viewset and then you use it. The same way you define a serializer and then you use it calling `get_serializer`, and so on.

Reply:
Yes, but you're setting it as none just after…

Reply:
Just like I can apply a `filter` or an `order_by` or whatever functions existing on this object. Both solutions are valid, I just thought getting the `queryset` using `get_queryset` was a better practice than using directly the model object. But maybe I'm wrong

Reply:
+1 on always using the queryset to get the object although I agree here the result is the same.

Reply:
Done